prediction_market_agent/tools/anvil/anvil_requests.py

```diff
@@ -20,3 +25,16 @@ def set_balance(rpc_url: str, address: str, balance: int) -> None:
     if "error" in response.json():
         raise ValueError(f"error occurred: {response.json()}")
     print(f"Set balance {balance} xDAI for address {address}")
+
+
+@contextlib.contextmanager
+def impersonate_account(
+    w3: Web3, account: ChecksumAddress
+) -> Generator[None, None, None]:
+    w3.provider.make_request(RPCEndpoint("anvil_impersonateAccount"), [account])
+    try:
+        yield
+    finally:
+        w3.provider.make_request(
+            RPCEndpoint("anvil_stopImpersonatingAccount"), [account]
+        )
```

Comment:
Impersonating means sending transactions on behalf of someone without someone's private key (possible only on Anvil)